dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/JdbcEventStore.java

```diff
@@ -603,141 +600,113 @@ lateral jsonb_each(
   }
 
   /**
-   * Generates a single INNER JOIN for each attribute we are filtering or ordering on. We can search
-   * by a range of operators. All searching is using lower() since attribute values are
-   * case-insensitive.
+   * Generates the WHERE-clause related to the user provided attribute filters. It will find the
+   * tracked entity attributes that match the given user filter criteria. This condition only
+   * applies when an attribute filter is specified.
    */
-  private String joinAttributeValue(
-      EventQueryParams params, MapSqlParameterSource mapSqlParameterSource) {
-    StringBuilder sql = new StringBuilder();
-
+  private String getWhereClauseFromAttributeFilterConditions(
+      EventQueryParams params, MapSqlParameterSource mapSqlParameterSource, SqlHelper hlp) {
+    StringBuilder fromBuilder = new StringBuilder();
     for (Entry<TrackedEntityAttribute, List<QueryFilter>> queryItem :
         params.getAttributes().entrySet()) {
-
       TrackedEntityAttribute tea = queryItem.getKey();
       String teaUid = tea.getUid();
       String teaValueCol = quote(teaUid);
-      String teaCol = quote(teaUid + "ATT");
 
-      sql.append(" inner join trackedentityattributevalue ")
-          .append(teaValueCol)
-          .append(" on ")
-          .append(teaValueCol + ".trackedentityid")
-          .append(" = TE.trackedentityid ")
-          .append(" inner join trackedentityattribute ")
-          .append(teaCol)
-          .append(" on ")
-          .append(teaValueCol + ".trackedentityattributeid")
-          .append(EQUALS)
-          .append(teaCol + ".trackedentityattributeid")
+      fromBuilder
+          .append(hlp.whereAnd())
+          .append(" TE.trackedentityid is not null ") // filters out results from event programs
```

Comment:
Is this filter automatically filtering out all the event programs if there is one attribute filter present in the request?
I mean, if we have a request asking for a null attribute, are we only going to return tracker events and filtering out all the program events?

Reply:
I'm working under the assumption that:

> An event program doesn’t have attributes, only data elements.

I arrived at this conclusion because the UI doesn’t provide an option to assign attributes to an event program, and when I query the database, event programs have no entries in `program_attributes` or `trackedentityattribute`.

Given this, I need to exclude event programs from the results when filtering by the `null` operator, as attributes are always null for event programs.